### PR TITLE
Nc games 9 get comments by review

### PR DESCRIPTION
**Pull request by jefner876**
add GET /api/reviews/:review_id/comments end point to get comments by review id

- check 0 comments lead to 200 status response
- handle invalid review_id
- handle review_id out of range

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -259,3 +259,57 @@ describe("/api/reviews", () => {
...
+          const { comments } = body;
+          expect(comments).toBeInstanceOf(Array);
+          expect(comments.length).toBe(3);
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id", expect.any(Number));
+            expect(comment).toHaveProperty("votes", expect.any(Number));
+            expect(comment).toHaveProperty("created_at", expect.any(String));
+            expect(isNaN(Date.parse(comment.created_at))).toBe(false); //expect string to be in date format
+            expect(comment).toHaveProperty("body", expect.any(String));
+            expect(comment).toHaveProperty("review_id", expect.any(Number));
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: You can be more specific with the review_id assertion as you know all these comments should belong to review_id 2

File: controllers/reviews.controller.js [hunk 2]
```diff
@@ -28,3 +29,20 @@ exports.getReviews = (req, res) => {
     res.status(200).send({ reviews });
   });
 };
+
+exports.getCommentsByReviewID = (req, res, next) => {
+  const { review_id: id } = req.params;
+  fetchCommentsByReviewID(id)
+    .then((comments) => {
+      if (comments.length === 0) {
```
Comment (Poonam-raj) [inline controllers/reviews.controller.js]: you should be checking if the request is a valid article with no comments vs a invalid article inside the model. The controller should just be using fetchCommentsByReviewID then sending a res or catching and nothing more really

tldr: keep the logic in your model

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Please move your logic to see if its a valid article with no comments associated or otherwise into the model and send another PR over :D 

Thanks

**Review by Poonam-raj — APPROVED** [review]
wonderful stuff!
